### PR TITLE
Allow custom labels to be specified for controller pods

### LINKED ISSUE #2951: gha-runner-scale-set-controller: supporting the custom labels on the pod template level
### What would you like added?

I would like to add the possibility to customise the labels on the pod template level similarly to how it's done on the annotations.

### Why is this needed?

Our organisation requires to set some labels on the pod level in order to apply some network policies. Without them the gatekeeper rejects the deployments.

The rationale behind is very simple: we can modify the annotations on the pod level as well as the labels on the deployment level, but these on the pod level cannot be changed without modifying the chart. It should be allowed IMHO. 

### Additional context

**Possible implementation**

`charts/gha-runner-scale-set-controller/templates/deployment.yaml`
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ include "gha-runner-scale-set-controller.fullname" . }}
  namespace: {{ .Release.Namespace }}
  labels:
    {{- include "gha-runner-scale-set-controller.labels" . | nindent 4 }}
    actions.github.com/controller-service-account-namespace: {{ .Release.Namespace }}
    actions.github.com/controller-service-account-name: {{ include "gha-runner-scale-set-controller.serviceAccountName" . }}
    {{- if .Values.flags.watchSingleNamespace }}
    actions.github.com/controller-watch-single-namespace: {{ .Values.flags.watchSingleNamespace }}
    {{- end }}
spec:
  replicas: {{ default 1 .Values.replicaCount }}
  selector:
    matchLabels:
      {{- include "gha-runner-scale-set-controller.selectorLabels" . | nindent 6 }}
  template:
    metadata:
      annotations:
        kubectl.kubernetes.io/default-container: "manager"
      {{- with .Values.podAnnotations }}
        {{- toYaml . | nindent 8 }}
      {{- end }}
      labels:
        app.kubernetes.io/part-of: gha-rs-controller
        app.kubernetes.io/component: controller-manager
        app.kubernetes.io/version: {{ .Chart.Version }}
        {{- include "gha-runner-scale-set-controller.selectorLabels" . | nindent 8 }}
        {{- with .Values.podTemplateLabels }}
          {{- toYaml . | nindent 8 }}
        {{- end }}
...
```
`charts/gha-runner-scale-set-controller/values.yaml`
```
# Default values for gha-runner-scale-set-controller.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.
labels: {}

podTemplateLabels: {}
...
```

### PR DESCRIPTION
**Pull request by wherka-ama**
Add ability to add custom labels to the controller pods.

Fixes #2951

### CODE REVIEW

**Comment by wherka-ama** [comment]
/fixes https://github.com/actions/actions-runner-controller/issues/2951

File: charts/gha-runner-scale-set-controller/values.yaml [hunk 1]
```diff
@@ -3,6 +3,8 @@
 # Declare variables to be passed into your templates.
 labels: {}
 
+podTemplateLabels: {}
```
Comment (nikola-jokic) [inline charts/gha-runner-scale-set-controller/values.yaml]: Could you move this below the podAnnotations and also name it `podLabels`?

Comment (wherka-ama) [reply charts/gha-runner-scale-set-controller/values.yaml]: Done. 
Thanks a lot @nikola-jokic !

Comment (nikola-jokic) [reply charts/gha-runner-scale-set-controller/values.yaml]: I'll run some tests just to make sure nothing breaks, but looks good to me! Thank you for your contribution!

**Review by nikola-jokic — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**